From 9eb35c67420fa5b1910e2e7e5d37ecef67a0e8b9 Mon Sep 17 00:00:00 2001
From: Sarai Hearthstone <shearthstone@habana.ai>
Date: Thu, 12 Jun 2025 08:57:22 +0000
Subject: [PATCH] Revert "Revert "[SW-224258] emb: pyhlml: call new api from old""

Problems on CD's side have been fixed, this change can once again be
merged.

This reverts commit 1c49ae07225e598ccf937e829e37add8b9cf575a.

Change-Id: Iedaf18dbc73daec58b42a6a3fae7aa949a2f344c
---

diff --git a/pyhlml/pyhlml/hlml_types.py b/pyhlml/pyhlml/hlml_types.py
index 3fb30d4..6a33f52 100644
--- a/pyhlml/pyhlml/hlml_types.py
+++ b/pyhlml/pyhlml/hlml_types.py
@@ -4,29 +4,31 @@
 
 import ctypes
 
+import pyhlml.pyhlml_types_generated as newpy
+
 class HLML_DEVICE:
     TYPE                                    = ctypes.c_void_p
 
 class HLML_DEFINE:
-    PCI_DOMAIN_LEN                          = 9
-    PCI_ADDR_LEN                            = ( PCI_DOMAIN_LEN + 10 )
-    PCI_LINK_INFO_LEN                       = 10
-    HL_FIELD_MAX_SIZE                       = 32
-    PORTS_ARR_SIZE                          = 2
-    HLML_DEVICE_ROW_RPL_MAX                 = 32
-    HLML_DEVICE_MAC_MAX_ADDRESSES           = 48
-    HLML_EVENT_ECC_ERR                      = ( 1 << 0 )
-    HLML_EVENT_ECC_DERR                     = ( 1 << 0 )
-    HLML_EVENT_CRITICAL_ERR                 = ( 1 << 1 )
-    HLML_EVENT_CLOCK_RATE                   = ( 1 << 2 )
-    HLML_EVENT_DRAM_ERR                     = ( 1 << 3 )
-    HLML_EVENT_DRAM_DERR                    = ( 1 << 3 )
-    HLML_EVENT_ECC_SERR                     = ( 1 << 4 )
-    HLML_EVENT_DRAM_SERR                    = ( 1 << 5 )
-    HLML_CLOCKS_THROTTLE_REASON_POWER       = ( 1 << 0 )
-    HLML_CLOCKS_THROTTLE_REASON_THERMAL     = ( 1 << 1 )
-    HLML_AFFINITY_SCOPE_NODE                = 0
-    HLML_AFFINITY_SCOPE_SOCKET              = 1
+    PCI_DOMAIN_LEN                          = newpy.HLML_DEFINES.PCI_DOMAIN_LEN
+    PCI_ADDR_LEN                            = newpy.HLML_DEFINES.PCI_ADDR_LEN
+    PCI_LINK_INFO_LEN                       = newpy.HLML_DEFINES.PCI_LINK_INFO_LEN
+    HL_FIELD_MAX_SIZE                       = newpy.HLML_DEFINES.HL_FIELD_MAX_SIZE
+    PORTS_ARR_SIZE                          = newpy.HLML_DEFINES.PORTS_ARR_SIZE
+    HLML_DEVICE_ROW_RPL_MAX                 = newpy.HLML_DEFINES.HLML_DEVICE_ROW_RPL_MAX
+    HLML_DEVICE_MAC_MAX_ADDRESSES           = newpy.HLML_DEFINES.HLML_DEVICE_MAC_MAX_ADDRESSES
+    HLML_EVENT_ECC_ERR                      = newpy.HLML_DEFINES.HLML_EVENT_ECC_ERR
+    HLML_EVENT_ECC_DERR                     = newpy.HLML_DEFINES.HLML_EVENT_ECC_DERR
+    HLML_EVENT_CRITICAL_ERR                 = newpy.HLML_DEFINES.HLML_EVENT_CRITICAL_ERR
+    HLML_EVENT_CLOCK_RATE                   = newpy.HLML_DEFINES.HLML_EVENT_CLOCK_RATE
+    HLML_EVENT_DRAM_ERR                     = newpy.HLML_DEFINES.HLML_EVENT_DRAM_ERR
+    HLML_EVENT_DRAM_DERR                    = newpy.HLML_DEFINES.HLML_EVENT_DRAM_DERR
+    HLML_EVENT_ECC_SERR                     = newpy.HLML_DEFINES.HLML_EVENT_ECC_SERR
+    HLML_EVENT_DRAM_SERR                    = newpy.HLML_DEFINES.HLML_EVENT_DRAM_SERR
+    HLML_CLOCKS_THROTTLE_REASON_POWER       = newpy.HLML_DEFINES.HLML_CLOCKS_THROTTLE_REASON_POWER
+    HLML_CLOCKS_THROTTLE_REASON_THERMAL     = newpy.HLML_DEFINES.HLML_CLOCKS_THROTTLE_REASON_THERMAL
+    HLML_AFFINITY_SCOPE_NODE                = newpy.HLML_DEFINES.HLML_AFFINITY_SCOPE_NODE
+    HLML_AFFINITY_SCOPE_SOCKET              = newpy.HLML_DEFINES.HLML_AFFINITY_SCOPE_SOCKET
 
 class COMMON_DEFINE:
     VERSION_MAX_LEN                         = 128
@@ -34,122 +36,126 @@
     HABANA_LINK_CNT_MAX_NUM                 = 256
     STATUS_MAX_LEN                          = 30
 
+# Duplicate values from "defines" class, kept for legacy reasons
 class HLML_AFFINITY_SCOPE:
     TYPE                                    = ctypes.c_uint
-    HLML_AFFINITY_SCOPE_NODE                = 0
-    HLML_AFFINITY_SCOPE_SOCKET              = 1
+    HLML_AFFINITY_SCOPE_NODE                = newpy.HLML_DEFINES.HLML_AFFINITY_SCOPE_NODE
+    HLML_AFFINITY_SCOPE_SOCKET              = newpy.HLML_DEFINES.HLML_AFFINITY_SCOPE_SOCKET
 
 class HLML_RETURN:
     TYPE                                    = ctypes.c_uint
-    HLML_SUCCESS                            = 0
-    HLML_ERROR_UNINITIALIZED                = 1
-    HLML_ERROR_INVALID_ARGUMENT             = 2
-    HLML_ERROR_NOT_SUPPORTED                = 3
-    HLML_ERROR_ALREADY_INITIALIZED          = 5
-    HLML_ERROR_NOT_FOUND                    = 6
-    HLML_ERROR_INSUFFICIENT_SIZE            = 7
-    HLML_ERROR_DRIVER_NOT_LOADED            = 9
-    HLML_ERROR_TIMEOUT                      = 10
-    HLML_ERROR_AIP_IS_LOST                  = 15
-    HLML_ERROR_MEMORY                       = 20
-    HLML_ERROR_NO_DATA                      = 21
-    HLML_ERROR_UNKNOWN                      = 49
+    HLML_SUCCESS                            = newpy.HLML_RETURN.HLML_SUCCESS
+    HLML_ERROR_UNINITIALIZED                = newpy.HLML_RETURN.HLML_ERROR_UNINITIALIZED
+    HLML_ERROR_INVALID_ARGUMENT             = newpy.HLML_RETURN.HLML_ERROR_INVALID_ARGUMENT
+    HLML_ERROR_NOT_SUPPORTED                = newpy.HLML_RETURN.HLML_ERROR_NOT_SUPPORTED
+    HLML_ERROR_ALREADY_INITIALIZED          = newpy.HLML_RETURN.HLML_ERROR_ALREADY_INITIALIZED
+    HLML_ERROR_NOT_FOUND                    = newpy.HLML_RETURN.HLML_ERROR_NOT_FOUND
+    HLML_ERROR_INSUFFICIENT_SIZE            = newpy.HLML_RETURN.HLML_ERROR_INSUFFICIENT_SIZE
+    HLML_ERROR_DRIVER_NOT_LOADED            = newpy.HLML_RETURN.HLML_ERROR_DRIVER_NOT_LOADED
+    HLML_ERROR_TIMEOUT                      = newpy.HLML_RETURN.HLML_ERROR_TIMEOUT
+    HLML_ERROR_AIP_IS_LOST                  = newpy.HLML_RETURN.HLML_ERROR_AIP_IS_LOST
+    HLML_ERROR_MEMORY                       = newpy.HLML_RETURN.HLML_ERROR_MEMORY
+    HLML_ERROR_NO_DATA                      = newpy.HLML_RETURN.HLML_ERROR_NO_DATA
+    HLML_ERROR_UNKNOWN                      = newpy.HLML_RETURN.HLML_ERROR_UNKNOWN
 
 class HLML_CLOCK_TYPE:
     TYPE                                    = ctypes.c_uint
-    HLML_CLOCK_SOC                          = 0
-    HLML_CLOCK_IC                           = 1
-    HLML_CLOCK_MME                          = 2
-    HLML_CLOCK_TPC                          = 3
-    HLML_CLOCK_COUNT                        = 4
+    HLML_CLOCK_SOC                          = newpy.HLML_CLOCK_TYPE.HLML_CLOCK_SOC
+    HLML_CLOCK_IC                           = newpy.HLML_CLOCK_TYPE.HLML_CLOCK_IC
+    HLML_CLOCK_MME                          = newpy.HLML_CLOCK_TYPE.HLML_CLOCK_MME
+    HLML_CLOCK_TPC                          = newpy.HLML_CLOCK_TYPE.HLML_CLOCK_TPC
+    HLML_CLOCK_COUNT                        = newpy.HLML_CLOCK_TYPE.HLML_CLOCK_COUNT
 
 class HLML_TEMP_SENS:
     TYPE                                    = ctypes.c_uint
-    HLML_TEMPERATURE_ON_AIP                 = 0
-    HLML_TEMPERATURE_ON_BOARD               = 1
-    HLML_TEMPERATURE_OTHER                  = 2
-    HLML_TEMPERATURE_HBM                    = 3
-    HLML_TEMPERATURE_VRM                    = 4
-    HLML_TEMPERATURE_CTEMP                  = 5
+    HLML_TEMPERATURE_ON_AIP                 = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_ON_AIP
+    HLML_TEMPERATURE_ON_BOARD               = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_ON_BOARD
+    HLML_TEMPERATURE_OTHER                  = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_OTHER
+    HLML_TEMPERATURE_HBM                    = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_HBM
+    HLML_TEMPERATURE_VRM                    = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_VRM
+    HLML_TEMPERATURE_CTEMP                  = newpy.HLML_TEMPERATURE_SENSORS.HLML_TEMPERATURE_CTEMP
 
 class HLML_TEMP_THRESH:
     TYPE                                    = ctypes.c_uint
-    HLML_TEMPERATURE_THRESHOLD_SHUTDOWN     = 0
-    HLML_TEMPERATURE_THRESHOLD_SLOWDOWN     = 1
-    HLML_TEMPERATURE_THRESHOLD_MEM_MAX      = 2
-    HLML_TEMPERATURE_THRESHOLD_GPU_MAX      = 3
-    HLML_TEMPERATURE_THRESHOLD_COUNT        = 4
+    HLML_TEMPERATURE_THRESHOLD_SHUTDOWN     = newpy.HLML_TEMPERATURE_THRESHOLDS.HLML_TEMPERATURE_THRESHOLD_SHUTDOWN
+    HLML_TEMPERATURE_THRESHOLD_SLOWDOWN     = newpy.HLML_TEMPERATURE_THRESHOLDS.HLML_TEMPERATURE_THRESHOLD_SLOWDOWN
+    HLML_TEMPERATURE_THRESHOLD_MEM_MAX      = newpy.HLML_TEMPERATURE_THRESHOLDS.HLML_TEMPERATURE_THRESHOLD_MEM_MAX
+    HLML_TEMPERATURE_THRESHOLD_GPU_MAX      = newpy.HLML_TEMPERATURE_THRESHOLDS.HLML_TEMPERATURE_THRESHOLD_GPU_MAX
+    HLML_TEMPERATURE_THRESHOLD_COUNT        = newpy.HLML_TEMPERATURE_THRESHOLDS.HLML_TEMPERATURE_THRESHOLD_COUNT
 
 class HLML_ENABLE_STATE:
     TYPE                                    = ctypes.c_uint
-    HLML_FEATURE_DISABLED                   = 0
-    HLML_FEATURE_ENABLED                    = 1
+    HLML_FEATURE_DISABLED                   = newpy.HLML_ENABLE_STATE.HLML_FEATURE_DISABLED
+    HLML_FEATURE_ENABLED                    = newpy.HLML_ENABLE_STATE.HLML_FEATURE_ENABLED
 
 class HLML_P_STATES:
     TYPE                                    = ctypes.c_uint
-    HLML_PSTATE_0                           = 0
-    HLML_PSTATE_1                           = 1
-    HLML_PSTATE_2                           = 2
-    HLML_PSTATE_3                           = 3
-    HLML_PSTATE_4                           = 4
-    HLML_PSTATE_5                           = 5
-    HLML_PSTATE_6                           = 6
-    HLML_PSTATE_7                           = 7
-    HLML_PSTATE_8                           = 8
-    HLML_PSTATE_9                           = 9
-    HLML_PSTATE_10                          = 10
-    HLML_PSTATE_11                          = 11
-    HLML_PSTATE_12                          = 12
-    HLML_PSTATE_13                          = 13
-    HLML_PSTATE_14                          = 14
-    HLML_PSTATE_15                          = 15
-    HLML_PSTATE_SENTINEL                    = 16
-    HLML_PSTATE_UNKNOWN                     = 32
+    HLML_PSTATE_0                           = newpy.HLML_P_STATES.HLML_PSTATE_0
+    HLML_PSTATE_1                           = newpy.HLML_P_STATES.HLML_PSTATE_1
+    HLML_PSTATE_2                           = newpy.HLML_P_STATES.HLML_PSTATE_2
+    HLML_PSTATE_3                           = newpy.HLML_P_STATES.HLML_PSTATE_3
+    HLML_PSTATE_4                           = newpy.HLML_P_STATES.HLML_PSTATE_4
+    HLML_PSTATE_5                           = newpy.HLML_P_STATES.HLML_PSTATE_5
+    HLML_PSTATE_6                           = newpy.HLML_P_STATES.HLML_PSTATE_6
+    HLML_PSTATE_7                           = newpy.HLML_P_STATES.HLML_PSTATE_7
+    HLML_PSTATE_8                           = newpy.HLML_P_STATES.HLML_PSTATE_8
+    HLML_PSTATE_9                           = newpy.HLML_P_STATES.HLML_PSTATE_9
+    HLML_PSTATE_10                          = newpy.HLML_P_STATES.HLML_PSTATE_10
+    HLML_PSTATE_11                          = newpy.HLML_P_STATES.HLML_PSTATE_11
+    HLML_PSTATE_12                          = newpy.HLML_P_STATES.HLML_PSTATE_12
+    HLML_PSTATE_13                          = newpy.HLML_P_STATES.HLML_PSTATE_13
+    HLML_PSTATE_14                          = newpy.HLML_P_STATES.HLML_PSTATE_14
+    HLML_PSTATE_15                          = newpy.HLML_P_STATES.HLML_PSTATE_15
+    HLML_PSTATE_SENTINEL                    = newpy.HLML_P_STATES.HLML_PSTATE_SENTINEL
+    HLML_PSTATE_UNKNOWN                     = newpy.HLML_P_STATES.HLML_PSTATE_UNKNOWN
 
-HLML_PSTATE_NUM_SUPPORTED = HLML_P_STATES.HLML_PSTATE_SENTINEL
+HLML_PSTATE_NUM_SUPPORTED = newpy.HLML_P_STATES.HLML_PSTATE_SENTINEL
 
 class HLML_MEMORY_ERROR:
     TYPE                                    = ctypes.c_uint
-    HLML_MEMORY_ERROR_TYPE_CORRECTED        = 0
-    HLML_MEMORY_ERROR_TYPE_UNCORRECTED      = 1
-    HLML_MEMORY_ERROR_TYPE_COUNT            = 2
+    HLML_MEMORY_ERROR_TYPE_CORRECTED        = newpy.HLML_MEMORY_ERROR_TYPE.HLML_MEMORY_ERROR_TYPE_CORRECTED
+    HLML_MEMORY_ERROR_TYPE_UNCORRECTED      = newpy.HLML_MEMORY_ERROR_TYPE.HLML_MEMORY_ERROR_TYPE_UNCORRECTED
+    HLML_MEMORY_ERROR_TYPE_COUNT            = newpy.HLML_MEMORY_ERROR_TYPE.HLML_MEMORY_ERROR_TYPE_COUNT
 
 class HLML_MEMORY_LOCATION:
     TYPE                                    = ctypes.c_uint
-    HLML_MEMORY_LOCATION_SRAM               = 0
-    HLML_MEMORY_LOCATION_DRAM               = 1
-    HLML_MEMORY_LOCATION_COUNT              = 2
+    HLML_MEMORY_LOCATION_SRAM               = newpy.HLML_MEMORY_LOCATION_TYPE.HLML_MEMORY_LOCATION_SRAM
+    HLML_MEMORY_LOCATION_DRAM               = newpy.HLML_MEMORY_LOCATION_TYPE.HLML_MEMORY_LOCATION_DRAM
+    HLML_MEMORY_LOCATION_ALL                = newpy.HLML_MEMORY_LOCATION_TYPE.HLML_MEMORY_LOCATION_ALL
+    HLML_MEMORY_LOCATION_COUNT              = newpy.HLML_MEMORY_LOCATION_TYPE.HLML_MEMORY_LOCATION_COUNT
 
 class HLML_ECC_COUNTER:
     TYPE                                    = ctypes.c_uint
-    HLML_VOLATILE_ECC                       = 0
-    HLML_AGGREGATE_ECC                      = 1
-    HLML_ECC_COUNTER_TYPE_COUNT             = 2
+    HLML_VOLATILE_ECC                       = newpy.HLML_ECC_COUNTER_TYPE.HLML_VOLATILE_ECC
+    HLML_AGGREGATE_ECC                      = newpy.HLML_ECC_COUNTER_TYPE.HLML_AGGREGATE_ECC
+    HLML_ECC_COUNTER_TYPE_COUNT             = newpy.HLML_ECC_COUNTER_TYPE.HLML_ECC_COUNTER_TYPE_COUNT
 
 class HLML_PCIE_UTIL_COUNTER:
     TYPE                                    = ctypes.c_uint
-    HLML_PCIE_UTIL_TX_BYTES                 = 0
-    HLML_PCIE_UTIL_RX_BYTES                 = 1
-    HLML_PCIE_UTIL_COUNT                    = 2
+    HLML_PCIE_UTIL_TX_BYTES                 = newpy.HLML_PCIE_UTIL_COUNTER.HLML_PCIE_UTIL_TX_BYTES
+    HLML_PCIE_UTIL_RX_BYTES                 = newpy.HLML_PCIE_UTIL_COUNTER.HLML_PCIE_UTIL_RX_BYTES
+    HLML_PCIE_UTIL_COUNT                    = newpy.HLML_PCIE_UTIL_COUNTER.HLML_PCIE_UTIL_COUNT
 
 class HLML_EVENT_SET:
     TYPE                                    = ctypes.c_void_p
 
 class HLML_ROW_REPLACEMENT_CAUSE:
     TYPE                                                      = ctypes.c_uint
-    HLML_ROW_REPLACEMENT_CAUSE_MULTIPLE_SINGLE_BIT_ECC_ERRORS = 0,
-    HLML_ROW_REPLACEMENT_CAUSE_DOUBLE_BIT_ECC_ERROR           = 1,
-    HLML_ROW_REPLACEMENT_CAUSE_COUNT                          = 2
+    HLML_ROW_REPLACEMENT_CAUSE_MULTIPLE_SINGLE_BIT_ECC_ERRORS = newpy.HLML_ROW_REPLACEMENT_CAUSE.HLML_ROW_REPLACEMENT_CAUSE_MULTIPLE_SINGLE_BIT_ECC_ERRORS
+    HLML_ROW_REPLACEMENT_CAUSE_DOUBLE_BIT_ECC_ERROR           = newpy.HLML_ROW_REPLACEMENT_CAUSE.HLML_ROW_REPLACEMENT_CAUSE_DOUBLE_BIT_ECC_ERROR
+    HLML_ROW_REPLACEMENT_CAUSE_COUNT                          = newpy.HLML_ROW_REPLACEMENT_CAUSE.HLML_ROW_REPLACEMENT_CAUSE_COUNT
 
 class HLML_PERF_POLICY:
     TYPE                                    = ctypes.c_uint
-    HLML_PERF_POLICY_POWER                  = 0,
-    HLML_PERF_POLICY_THERMAL                = 1,
-    HLML_PERF_POLICY_COUNT                  = 0
+    HLML_PERF_POLICY_POWER                  = newpy.HLML_PERF_POLICY_TYPE.HLML_PERF_POLICY_POWER
+    HLML_PERF_POLICY_THERMAL                = newpy.HLML_PERF_POLICY_TYPE.HLML_PERF_POLICY_THERMAL
+    HLML_PERF_POLICY_COUNT                  = newpy.HLML_PERF_POLICY_TYPE.HLML_PERF_POLICY_COUNT
 
+# Legacy
 class _struct_c_hlml_unit(ctypes.Structure):
     pass # opaque handle
 
+# Legacy
 class HLML_UNIT:
     TYPE                                    = _struct_c_hlml_unit
 
@@ -192,69 +198,31 @@
         return self.__class__.__name__ + "(" +  ", ".join(result) + ")"
 
 class c_hlml_pci_cap(_PrintS):
-    _fields_ = [("link_speed", ctypes.c_char * HLML_DEFINE.PCI_LINK_INFO_LEN),
-                ("link_width",ctypes.c_char * HLML_DEFINE.PCI_LINK_INFO_LEN),
-                ("link_max_speed", ctypes.c_char * HLML_DEFINE.PCI_LINK_INFO_LEN),
-                ("link_max_width",ctypes.c_char * HLML_DEFINE.PCI_LINK_INFO_LEN)
-               ]
+    _fields_ = newpy.c_hlml_pci_cap._fields_
 
 class c_hlml_pci_info(_PrintS):
-    """
-    /*
-        * bus - The bus on which the device resides, 0 to 0xf
-        * bus_id - The tuple domain:bus:device.function
-        * device - The device's id on the bus, 0 to 31
-        * domain - The PCI domain on which the device's bus resides
-        * pci_device_id - The combined 16b deviceId and 16b vendor id
-        * pci_subsys_device_id - The combined 16b subsys_id and 16b subsys_vendor_id
-    */
-    """
-    _fields_ = [("bus", ctypes.c_uint),
-                ("bus_id", ctypes.c_char * HLML_DEFINE.PCI_ADDR_LEN),
-                ("device", ctypes.c_uint),
-                ("domain", ctypes.c_uint),
-                ("pci_device_id", ctypes.c_uint),
-                ("caps", c_hlml_pci_cap),
-                ("pci_rev", ctypes.c_uint),
-                ("pci_subsys_id", ctypes.c_uint)
-               ]
+    _fields_ = newpy.c_hlml_pci_info._fields_
 
 class c_hlml_utilization(_PrintS):
-    _fields_ = [("aip", ctypes.c_uint),
-                ("memory", ctypes.c_uint)
-               ]
+    _fields_ = newpy.c_hlml_utilization._fields_
 
 class c_hlml_process_utilization(_PrintS):
-    _fields_ = [("aip_util", ctypes.c_uint)
-               ]
+    _fields_ = newpy.c_hlml_process_utilization_sample._fields_
 
 class c_hlml_memory(_PrintS):
-    _fields_ = [("free", ctypes.c_ulonglong),
-                ("total", ctypes.c_ulonglong),
-                ("used", ctypes.c_ulonglong)
-               ]
+    _fields_ = newpy.c_hlml_memory._fields_
 
 class c_hlml_pcb_info(_PrintS):
-    _fields_ = [("pcb_ver", ctypes.c_char * HLML_DEFINE.HL_FIELD_MAX_SIZE),
-                ("pcb_assembly_ver", ctypes.c_char * HLML_DEFINE.HL_FIELD_MAX_SIZE)
-               ]
+    _fields_ = newpy.c_hlml_pcb_info._fields_
 
 class c_hlml_event_data(_PrintS):
-    _fields_ = [("device", ctypes.c_void_p),
-                ("event_type", ctypes.c_ulonglong)
-               ]
+    _fields_ = newpy.c_hlml_event_data._fields_
 
 class c_hlml_mac_info(_PrintS):
-    _fields_ = [("addr", ctypes.c_ubyte * COMMON_DEFINE.ETHER_ADDR_LEN), # unsigned char
-                ("id", ctypes.c_int)
-               ]
+    _fields_ = newpy.c_hlml_mac_info._fields_
 
 class c_hlml_nic_stats_info(_PrintS):
-    _fields_ = [("port", ctypes.c_uint32),
-                ("str_buf", ctypes.POINTER(ctypes.c_char)),
-                ("val_buf", ctypes.POINTER(ctypes.c_uint64)),
-                ("num_of_counters_out", ctypes.POINTER(ctypes.c_uint32))
-                ]
+    _fields_ = newpy.c_hlml_nic_stats_info._fields_
 
     def __init__(self, port: int, num_of_counters: int = None):
         num_of_counters = num_of_counters or COMMON_DEFINE.HABANA_LINK_CNT_MAX_NUM
@@ -269,18 +237,12 @@
         self.num_of_counters_out = (ctypes.c_uint32 * 1)()
 
 class c_hlml_violation_time(_PrintS):
-    _fields_ = [("reference_time", ctypes.c_ulonglong),
-                ("violation_time", ctypes.c_ulonglong)
-               ]
+    _fields_ = newpy.c_hlml_violation_time._fields_
 
 class c_hlml_row_address(_PrintS):
-    _fields_ = [("hbm_idx", ctypes.c_uint8),
-                ("pc", ctypes.c_uint8),
-                ("sid", ctypes.c_uint8),
-                ("bank_idx", ctypes.c_uint8),
-                ("row_addr", ctypes.c_uint16)
-               ]
+    _fields_ = newpy.c_hlml_row_address._fields_
 
+# Legacy
 class hlml_ecc_mode(_PrintS):
     _fields_ = [("current", ctypes.c_uint),
                 ("pending", ctypes.c_uint)
